kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/HttpLoggingInterceptor.java

```diff
@@ -0,0 +1,206 @@
+// - Response
+//   Handled in the after method
+// - Response body
+//   Handled in by the consumer
+// For WS requests, we might need to look into further improving the StandardHttpClient implementation
+public class HttpLoggingInterceptor implements Interceptor {
+
+  private final HttpLogger httpLogger;
+  private final Map<UUID, DeferredLoggingConsumer> activeConsumers;
+
+  public HttpLoggingInterceptor() {
+    this(LoggerFactory.getLogger(HttpLoggingInterceptor.class));
+  }
+
+  public HttpLoggingInterceptor(Logger logger) {
+    this.httpLogger = new HttpLogger(logger);
+    activeConsumers = new ConcurrentHashMap<>();
+  }
+
+  @Override
+  public AsyncBody.Consumer<List<ByteBuffer>> consumer(AsyncBody.Consumer<List<ByteBuffer>> consumer, HttpRequest request) {
+    final DeferredLoggingConsumer interceptedConsumer = new DeferredLoggingConsumer(httpLogger, request, consumer,
+        () -> activeConsumers.remove(request.id()));
+    activeConsumers.put(request.id(), interceptedConsumer);
+    return interceptedConsumer;
+  }
+
+  @Override
+  public void after(HttpRequest request, HttpResponse<?> response) {
+    if (response instanceof WebSocketUpgradeResponse) {
+      httpLogger.logWsStart();
+      httpLogger.logRequest(request);
+      httpLogger.logResponse(response);
+      httpLogger.logWsEnd();
+    } else {
+      activeConsumers.computeIfPresent(request.id(), (id, consumer) -> {
+        if (response.body() instanceof AsyncBody) {
+          consumer.processAsyncBody((AsyncBody) response.body());
+        }
+        consumer.originalResponse.set(response);
+        return consumer;
+      });
+    }
+  }
+
+  private static final class DeferredLoggingConsumer
+      implements AsyncBody.Consumer<List<ByteBuffer>>, BiConsumer<Void, Throwable> {
+
+    private final HttpLogger httpLogger;
+    private final HttpRequest originalRequest;
+    private final AsyncBody.Consumer<List<ByteBuffer>> originalConsumer;
+    private final Runnable cleanUp;
+    private final AtomicReference<HttpResponse<?>> originalResponse;
+    private final AtomicBoolean asyncBodyDoneProcessed;
+    private final Queue<ByteBuffer> responseBody;
+
+    public DeferredLoggingConsumer(HttpLogger httpLogger, HttpRequest originalRequest,
+        AsyncBody.Consumer<List<ByteBuffer>> originalConsumer, Runnable cleanUp) {
+      this.httpLogger = httpLogger;
+      this.originalRequest = originalRequest;
+      this.originalConsumer = originalConsumer;
+      this.cleanUp = cleanUp;
+      originalResponse = new AtomicReference<>();
+      asyncBodyDoneProcessed = new AtomicBoolean(false);
+      responseBody = new ConcurrentLinkedQueue<>();
+    }
+
+    @Override
+    public void consume(List<ByteBuffer> value, AsyncBody asyncBody) throws Exception {
+      // TODO: Should try to detect if the body is text or not? (HttpLoggingInterceptor.isPlainText)
+      value.stream().map(BufferUtil::copy).forEach(responseBody::add); // Potential leak
```

Comment:
Yes doing something like a long running log watch, an http watch, etc. would indefinitely build up here.  This will need to be bounded in some way - time, size, or both.